lib/macho/macho_file.rb

```diff
@@ -305,23 +305,32 @@ def dylib_id
 
     # Changes the Mach-O's dylib ID to `new_id`. Does nothing if not a dylib.
     # @example
-    #  file.dylib_id = "libFoo.dylib"
+    #  file.change_dylib_id("libFoo.dylib")
     # @param new_id [String] the dylib's new ID
+    # @param options [Hash]
+    # @option options [Boolean] :strict (true) whether or not to fail loudly
+    #  with an exception if the change cannot be performed
     # @return [void]
     # @raise [ArgumentError] if `new_id` is not a String
-    def dylib_id=(new_id)
+    def change_dylib_id(new_id, options = {})
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
       old_lc = command(:LC_ID_DYLIB).first
-      raise DylibIdMissingError.new unless old_lc
+
+      if old_lc.nil?
+        return unless options.fetch(:strict, true)
+        raise DylibIdMissingError.new
+      end
```

Comment:
Sounds good, for this and all below. I'll also remove the `@option` documentation line now that we don't consider `strict` here.